src/cbmc_starter_kit/update.py

```diff
@@ -0,0 +1,211 @@
+                              f'CBMC root {args.cbmc_root}')
+    else:
+        args.no_migrate = True
+    logging.debug('CBMC starter kit root: %s', args.starter_kit_root)
+    return args
+
+################################################################
+
+def files_under_root(root=None, symlinks_only=False):
+    if root and not root.is_dir():
+        logging.critical('Not a directory: %s', root)
+        sys.exit(1)
+
+    cmd = ['find', '.']
+    if symlinks_only:
+        cmd += ['-type', 'l']
+    kwds = {
+        'cwd': root,
+        'text': True,
+        'capture_output': True,
+    }
+    result = subprocess.run(cmd, **kwds, check=True)
+
+    if not result.returncode:
+        return sorted(result.stdout.splitlines())
+    logging.critical('This should be impossible: Failed to list files under directory: %s', root)
+    sys.exit(1)
+
+################################################################
+
+def migrate(cbmc_root, starter_kit_root):
+    logging.debug('Migrating CBMC starter kit')
+    templates = files_under_root(starter_kit_root/util.REPOSITORY_TEMPLATES)
+    logging.debug('Migrating CBMC starter kit: found templates: %s', templates)
+    symlinks = files_under_root(cbmc_root, symlinks_only=True)
+    logging.debug('Migrating CBMC starter kit: found symlinks: %s', symlinks)
+
+    cbmc_symlinks = [path for path in templates if path in symlinks]
+    for symlink in cbmc_symlinks:
+        cbmc_link = cbmc_root / symlink
+        cbmc_file = cbmc_link.resolve()
+        logging.warning('Copying: %s -> %s', cbmc_file, cbmc_link)
+        assert cbmc_file.exists()
+        assert cbmc_link.exists()
+        cbmc_link.unlink()
+        shutil.copy(cbmc_file, cbmc_link)
+
+def remove_negative_tests(cbmc_root):
+    logging.debug('Removing CBMC starter kit negative tests')
+    negative_tests = cbmc_root / util.NEGATIVE_TESTS
+    if negative_tests.is_dir():
+        logging.warning('Removing: %s', negative_tests)
+        shutil.rmtree(negative_tests)
+
+def update(cbmc_root, quiet=False):
+    logging.debug('Updating CBMC starter kit')
+    for path in [f'{util.PROOF_DIR}/{util.COMMON_MAKEFILE}', f'{util.PROOF_DIR}/{util.RUN_SCRIPT}']:
+        src = util.package_repository_template_root() / path
+        dst = cbmc_root / path
+        (logging.debug if quiet else logging.warning)('Copying: %s -> %s', src, dst)
+        version.copy_with_version(src, dst)
+
+def remove_submodule(cbmc_root, submodule_name, submodule_path, force=False):
+    logging.debug('Checking for %s submodule', submodule_name)
+    if not submodule_path:
+        logging.debug('%s submodule does not exist', submodule_name)
+        return
+    if not force:
+        logging.warning('Use --force if you want to remove the %s submodule: %s',
+                        submodule_name, submodule_path)
+        return
+    for submodule in git.Repo(cbmc_root, search_parent_directories=True).submodules:
+        if submodule.path == str(submodule_path):
```

Comment:
Since `submodule_path` is already a `Path` object, could we compare two path objects instead of raw strings:
```suggestion
        if Path(submodule.path) == submodule_path:
```
so we don't have to worry about tricky cases like one of the paths being relative and other being absolute.

Reply:
Left as if after phone discussion concluding that Path did not resolve relative paths without invoking the resolve() method.